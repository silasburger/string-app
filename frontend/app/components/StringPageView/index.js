/**
 *
 * StringPageView
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import PostList from 'components/PostList';
import StringFormContainer from 'containers/StringFormContainer/Loadable';
import ErrorAlert from 'components/ErrorAlert';

/**
 * Errors are rendered in the component when they're present.
 * Disconnect the backend and give it a try.
 */

/* eslint-disable react/prefer-stateless-function */
export class StringPageView extends React.Component {
  render() {
    return (
      <div id="string-page">
        <Helmet>
          <title>String Page</title>
          <meta name="String Page" content="This page shows all the strings!" />
        </Helmet>
        {this.props.error !== false ? (
          <ErrorAlert error={this.props.error} />
        ) : null}
        <StringFormContainer />
        <PostList
          loading={this.props.loading}
          changeDateTimeFilter={this.props.changeDateTimeFilter}
          posts={this.props.posts}
          filter={this.props.filter}
        />
      </div>
    );
  }
}

StringPageView.propTypes = {
  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  posts: PropTypes.oneOfType([PropTypes.array, PropTypes.bool]),
  changeDateTimeFilter: PropTypes.func,
  loading: PropTypes.bool,
  filter: PropTypes.string,
};

export default StringPageView;
