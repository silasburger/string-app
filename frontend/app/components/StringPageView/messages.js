/*
 * StringPageView Messages
 *
 * This contains all the text for the StringPageView component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.StringPageView';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the StringPageView component!',
  },
});
