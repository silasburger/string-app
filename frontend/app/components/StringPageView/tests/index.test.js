import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StringPageView from '../index';

describe('<StringPageView />', () => {
  it('renders without crashing', () => {
    shallow(
      <StringPageView
        loading={false}
        error={false}
        posts={[]}
        filter="ALL_POSTS"
        changeDateTimeFilter={() => {}}
      />,
    );
  });
  it('matches snapshot', () => {
    const wrapper = shallow(
      <StringPageView
        loading={false}
        error={false}
        posts={[]}
        filter="ALL_POSTS"
        changeDateTimeFilter={() => {}}
      />,
    );
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
