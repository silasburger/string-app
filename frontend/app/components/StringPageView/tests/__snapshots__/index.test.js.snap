// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringPageView /> matches snapshot 1`] = `
<StringPageView__View>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      String Page
    </title>
    <meta
      content="This page shows all the strings!"
      name="String Page"
    />
  </HelmetWrapper>
  <LoadableComponent />
  <PostList
    changeDateTimeFilter={[Function]}
    filter="ALL_POSTS"
    loading={false}
    posts={Array []}
  />
</StringPageView__View>
`;
